app/controllers/visualizations_controller.rb

```diff
@@ -63,11 +63,14 @@ def save
     # Delete to have single source of truth.
     @data.delete(:sampleIds)
 
-    vis = Visualization.joins(:samples).where(
-      user: current_user,
-      visualization_type: @type,
-      samples: { id: [sample_ids] }
-    ).order(created_at: :desc)
+    vis = Visualization.joins(:samples)
+                       .where(
+                         user: current_user,
+                         visualization_type: @type,
+                         samples: { id: [sample_ids] }
+                       )
+                       .where.not(visualization_type: [nil, 'undefined'], name: nil)
```

Comment:
for my own education, when would each of the scenarios `visualization_type: [nil, 'undefined']` arise?

Reply:
I intended this only for filtering out early test data which did not have those values set. After this PR, they should always be set. 

I was going to add this as a comment but ruby does not seem to accept a # comment on a line in between method calls 🙁 

In any case, an alternative would be to delete those rows from the staging and prod DBs. How does the team feel about direct DB mutations?

Reply:
I think we're ok with direct mutations, particularly if they only involve derived or non-critical data. However, we want them to be always documented, reviewable and reproducible as explicit migrations :D